### ISSUE TITLE
AWS S3 [503 Slow Down] terminates expire command with exception [104]

### ISSUE THREAD

**Issue by cul-bsi** [pgbackrest/pgbackrest#2040]
1. pgBackRest version:
2.44-1.pgdg100+1

2. PostgreSQL version:
14.7

4. Operating system/version:
AWS EKS 1.24 / container image is based upon `debian:bullseye-slim`

5. Did you install pgBackRest from source or from a package?
apt package from http://apt.postgresql.org/pub/repos/apt

6. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
[pgbackrest.conf.gz](https://github.com/pgbackrest/pgbackrest/files/11133782/pgbackrest.conf.gz)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
N/A
    - errors in the postgresql log file before or during the time you experienced the issue
N/A
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)
[pgbackrest-error.log.gz](https://github.com/pgbackrest/pgbackrest/files/11133781/pgbackrest-error.log.gz)

7. Describe the issue:
After a successful `pgbackrest backup` command the `pgbackrest expire` command may fail with exit code 104 if it receives a `503 Slow Down` response from AWS S3 endpoint.
```
2023-04-02 01:10:48.738 P00   INFO: repo1: expire time-based backup 20230325-010012F
2023-04-02 01:10:48.817 P00   INFO: repo1: remove expired backup 20230325-010012F
2023-04-02 01:11:52.113 P00   INFO: expire command end: aborted with exception [104]
ERROR: [101]: repo1: [503] Slow Down
ERROR: [104]: expire command encountered 1 error(s), check the log file for details
command terminated with exit code 104
```

There may be hundreds of smaller (<10 GB) databases trying cleaning up their expired backups at the same time. If we restart the expire command at a later time, one by one, it usually works.

This only happens during `expire`, never during `backup`.

According to https://github.com/pgbackrest/pgbackrest/issues/622 this was fixed back in 2.07.

**Comment by dwsteele** [comment]
> This only happens during expire, never during backup.

Backup creates files more slowly than expire removes them (1000 at a time).

> According to https://github.com/pgbackrest/pgbackrest/issues/622 this was fixed back in 2.07.

This was fixed. You are seeing this error after multiple retries over `io-timeout` seconds, see https://github.com/pgbackrest/pgbackrest/blob/release/2.45/src/common/io/http/request.c#L155.

The best thing to do here is enable file bundling: https://pgbackrest.org/user-guide.html#backup/bundle. This greatly reduces the number of files to be deleted (and makes backup more efficient as well).

**Comment by dwsteele** [comment]
I have also added retry details to HTTP errors to make the underlying behavior a little clearer: f33e1cfb.
